### DISCUSSION TITLE
Information entropy and the size of the data

### DISCUSSION THREAD

**Discussion by Takodachi6969** [Part-II-Computational-Physics/.github#100] [General]
When I was doing Lenia I calculated the information entropy of each individual animal and the size of the animals. When I compared them they lined in a perfect straight line with Pearson correlation index of 0.99. I thought it was due to the way shannon entropy was computed via scipy.entropy calculation overlaped with the sum of non-zero squared but after some research online they said the size of the data should have had no corelation with the information entropy the data possess. ie an array from 1-10 and another array from 1-100 should have the same information entropy (which they did), but the perfect straight was too good to be true. Am I missing something?

**Comment by AustenLamacraft** [comment]
You might need to give me a bit more detail here, but entropy is in general an extensive quantity, meaning that you expect it to increase in proportion to the system size. What you've read may correspond to the entropy normalized by the system size...

**Reply by Takodachi6969** [reply]
The information entropy was measured by separating the lifeform's cells into bins in a histogram, then the probability distribution was calculated. Information entropy was calculated using the Shannon entropy in information theory which was defined as \sum P_i \ln{P_i}.

I have tried to test the extensiveness of this quantity, but then the final results were purely dependent on its probability distribution function, and again an array of size 10 and 100 output the same results. 

Which was why I was super confused with this perfect correlation

**Reply by Takodachi6969** [reply]
> You might need to give me a bit more detail here, but entropy is in general an extensive quantity, meaning that you expect it to increase in proportion to the system size. What you've read may correspond to the entropy normalized by the system size...

Here are the exact code I used to compute them

```
from scipy.stats import entropy

def normalize(animal_configuration):
    return (animal_configuration - np.min(animal_configuration)) / (np.max(animal_configuration) - np.min(animal_configuration))

def create_histogram(animal_configuration, num_bins=100):
    hist, _ = np.histogram(animal_configuration, bins=num_bins, range=(0, 1))
    return hist

def probability_distribution(hist):
    N_total = np.sum(hist)
    return hist / N_total

def information_entropy(prob_distribution):
    H = entropy(prob_distribution)
    return H


def calculate_size(arr):
    return np.count_nonzero(arr)
```

where the returned value of calculate_size and information_entropy correlated perfectly....

**Reply by AustenLamacraft** [reply]
Ah, I get it. You are looking at the entropy of the overall (normalized) pixel distribution. The maximum entropy you can get from this calculation is $\log N_\text{bins}$ corresponding to a uniform distribution in the $N_\text{bins}$ bins. 

Maybe as you increase the size you are including a smaller fraction of zero cells (not counted in your size calculation) and this is changing the pixel distribution

**Reply by Takodachi6969** [reply]
I did look into this, but in the file provided by Bert (The guy who made Lenia), the array that described the lifeform was essentially the smallest rectangle that fitted all the cells of the animal in it so I don't think the amount of zero cells changed by a lot to that. 

I also did a quick test to prove it, it seems that they are very similar. The zeros are sometimes distinct features of the animal (if they have holes within them)
<img width="668" alt="Screenshot 2023-04-15 184807" src="https://user-images.githubusercontent.com/102265625/232209444-92a98483-e777-4deb-96da-ad3229471393.png">
